compiler/noirc_frontend/src/ast/expression.rs

```diff
@@ -71,6 +72,21 @@ impl UnresolvedGeneric {
             }
         }
     }
+
+    pub fn kind(&self) -> Kind {
+        match self {
+            UnresolvedGeneric::Variable(_) => Kind::Normal,
+            // The inner numeric type of the kind cannot be resolved from an `UnresolvedGeneric`
+            // on its own. It is expected that the inner type is updated during elaboration.
+            UnresolvedGeneric::Numeric { .. } => Kind::Numeric { typ: Box::new(Type::Error) },
```

Comment:
I now return a result from `kind` which will return the unresolved type which we failed to resolve for the kind.

Reply:
Inside of the Context's `resolve_generics` method I still return a bad numeric generic with a `Type::Error`. This is because we continue with elaboration after definition collection and I didn't want to remove the `assert_eq!(unresolved_generics.len(), generics.len());` check inside of `add_existing_generics`